src/Datagrid/Pager.php

```diff
@@ -117,15 +115,22 @@ public function getNbResults()
 
     public function init()
     {
-        $this->resetIterator();
+        // NEXT_MAJOR: Remove this line.
+        $this->resetIterator('sonata_deprecation_mute');
 
         $this->setResultsCount($this->computeNbResult('sonata_deprecation_mute'));
 
         $this->getQuery()->setFirstResult(null);
         $this->getQuery()->setMaxResults(null);
 
-        if (\count($this->getParameters()) > 0) {
-            $this->getQuery()->setParameters($this->getParameters());
+        // NEXT_MAJOR: Remove this code.
+        if (\count($this->getParameters('sonata_deprecation_mute')) > 0) {
+            @trigger_error(
```

Comment:
I guess this would need an upgrade note (BTW we missed the upgrade note for https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1257, I'll add it to MongoDB when creating a similar PR to this)

Reply:
I don't think an upgrade note is needed here.

`setParameters` and all similar method are already deprecated by SonataAdmin, so you should not use them anymore.
Here we're just saying that we won't use them, to respect the SonataAdmin deprecation.

If someone use them, he would already have got the deprecation from SonataAdmin.

Reply:
So maybe triggering a new deprecation it is not needed. 

In general, if we trigger a new deprecation, the upgrade note should be updated.

Reply:
Done